apps/demos/Demos/CardView/DataValidation/jQuery/index.js

```diff
@@ -1,7 +1,148 @@
 $(() => {
   $('#card-view').dxCardView({
-    dataSource: customers,
-    keyExpr: 'ID',
-    columns: ['CompanyName', 'City', 'State', 'Phone', 'Fax'],
+    dataSource: employees,
+    keyExpr: 'id',
+    cardCover: {
+      imageExpr: ({ picture }) => picture,
+      altExpr: ({ fullName }) => `Photo of ${fullName}`,
+    },
+    editing: {
+      allowAdding: true,
+      allowUpdating: true,
+      allowDeleting: true,
+      popup: {
+        title: 'Employee Info',
+        showTitle: true,
+        width: 700,
+        height: 525,
+      },
+      form: {
+        items: [
+          {
+            caption: 'Personal Data',
+            itemType: 'group',
+            colCount: 2,
+            colSpan: 2,
+            items: ['firstName', 'lastName', 'birthDate', 'picture'],
+          }, {
+            caption: 'Main Info',
+            itemType: 'group',
+            colCount: 2,
+            colSpan: 2,
+            items: ['hireDate', 'title', {
+              dataField: 'notes',
+              editorType: 'dxTextArea',
+              colSpan: 2,
+              editorOptions: {
+                height: 100,
+              },
+            }],
+          }, {
+            caption: 'Contacts',
+            itemType: 'group',
+            colCount: 2,
+            colSpan: 2,
+            items: [
+              {
+                dataField: 'address',
+                colSpan: 2,
+              }, 'city', 'zipcode', 'mobilePhone', 'email',
+            ],
+          },
+        ],
+      },
+    },
+    searchPanel: {
+      visible: true,
+    },
+    columns: [
+      {
+        caption: 'Full Name',
+        calculateFieldValue({ firstName, lastName }) {
+          return `${firstName} ${lastName}`;
+        },
+      },
+      {
+        dataField: 'birthDate',
+        validationRules: [{ type: 'required' }],
+        dataType: 'date',
+      },
+      {
+        dataField: 'hireDate',
+        validationRules: [
+          { type: 'required' },
+          {
+            type: 'custom',
+            message: 'Hire date cannot be earlier than birth date',
+            validationCallback(params) {
+              return new Date(params.data.hireDate) > new Date(params.data.birthDate);
```

Comment:
During the validation, the data isn't applied 
In other words, before data changes applying the validation callback is called

So, the ```params.data.hireDate``` is undefined here (or previous value)
To fix related issues, let's change this line the following way:
```suggestion
              return new Date(params.value) > new Date(params.data.birthDate);
```

Also, let's change it in other frameworks :)

Reply:
It's a bug, it should work. It doesn't affect demo functionality, so let's fix separately

Reply:
Unfortunately, it works incorrectly when adding a new card 
E.g:
![2242EA53-D95F-413F-BD88-729E12FD4765](https://github.com/user-attachments/assets/a6174721-dbbd-475e-aa87-ba831a9efefb)

So, let's temporary use WA with ```params.value``` (maybe left ```TODO``` to fix demo code later) or block this demo by known bug

Reply:
used `params.value`